teamcity: failed tests on master: testrace (proposer evaluated kv)/TestBackupAsOfSystemTime

The following tests appear to have failed:

[#170441](https://teamcity.cockroachdb.com/viewLog.html?buildId=170441):

```
--- FAIL: testrace (proposer evaluated kv)/TestBackupAsOfSystemTime (9.630s)
<autogenerated>:14: error executing 'RESTORE DATABASE bench FROM '/go/src/github.com/cockroachdb/cockroach/artifacts/TestBackupAsOfSystemTime999567012'': pq: importing 10 ranges: first of 1 errors: import can only be called on empty ranges
------- Stdout: -------
W170306 17:26:42.206736 23865 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170306 17:26:42.212559 23865 server/config.go:482  1 storage engine initialized
I170306 17:26:42.213562 23865 server/node.go:443  [n?] store [n0,s0] not bootstrapped
I170306 17:26:42.226952 23865 server/node.go:372  [n?] **** cluster 28e50305-28c0-43d3-9e92-a9ea0199a477 has been created
I170306 17:26:42.227051 23865 server/node.go:373  [n?] **** add additional nodes by specifying --join=127.0.0.1:33637
I170306 17:26:42.231886 23865 storage/store.go:1274  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170306 17:26:42.232017 23865 server/node.go:456  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:0}
I170306 17:26:42.232201 23865 server/node.go:341  [n1] node ID 1 initialized
I170306 17:26:42.232389 23865 gossip/gossip.go:293  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33637" > attrs:<> locality:<> 
I170306 17:26:42.233120 23865 storage/stores.go:296  [n1] read 0 node addresses from persistent storage
I170306 17:26:42.233305 23865 server/node.go:605  [n1] connecting to gossip network to verify cluster ID...
I170306 17:26:42.234428 23865 server/node.go:629  [n1] node connected via gossip and verified as part of cluster "28e50305-28c0-43d3-9e92-a9ea0199a477"
I170306 17:26:42.234850 23865 server/node.go:391  [n1] node=1: started with [[]=] engine(s) and attributes []
I170306 17:26:42.235321 23865 sql/executor.go:337  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:33637}
I170306 17:26:42.247405 23865 server/server.go:651  [n1] starting https server at 127.0.0.1:47386
I170306 17:26:42.247515 23865 server/server.go:652  [n1] starting grpc/postgres server at 127.0.0.1:33637
I170306 17:26:42.247562 23865 server/server.go:653  [n1] advertising CockroachDB node at 127.0.0.1:33637
I170306 17:26:42.249070 23883 storage/split_queue.go:98  [split,n1,s1,r1/1:/M{in-ax},@c420aa7880] splitting at key /Table/0/0
I170306 17:26:42.262802 23883 storage/replica_command.go:2456  [split,n1,s1,r1/1:/M{in-ax},@c420aa7880] initiating a split of this range at key /Table/0 [r2]
I170306 17:26:42.339090 23883 storage/split_queue.go:98  [split,n1,s1,r2/1:/{Table/0-Max},@c422c0ae00] splitting at key /Table/11/0
I170306 17:26:42.339305 23883 storage/replica_command.go:2456  [split,n1,s1,r2/1:/{Table/0-Max},@c422c0ae00] initiating a split of this range at key /Table/11 [r3]
I170306 17:26:42.359620 23972 sql/event_log.go:95  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:33637} Attrs: Locality:} ClusterID:28e50305-28c0-43d3-9e92-a9ea0199a477 StartedAt:1488821202234486220 LastUp:1488821202234486220}
I170306 17:26:42.385657 23865 sql/event_log.go:95  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN uniqueID SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170306 17:26:42.398236 23883 storage/split_queue.go:98  [split,n1,s1,r3/1:/{Table/11-Max},@c422a78700] splitting at key /Table/12/0
I170306 17:26:42.398472 23883 storage/replica_command.go:2456  [split,n1,s1,r3/1:/{Table/11-Max},@c422a78700] initiating a split of this range at key /Table/12 [r4]
I170306 17:26:42.483749 23883 storage/split_queue.go:98  [split,n1,s1,r4/1:/{Table/12-Max},@c423bd9c00] splitting at key /Table/13/0
I170306 17:26:42.483971 23883 storage/replica_command.go:2456  [split,n1,s1,r4/1:/{Table/12-Max},@c423bd9c00] initiating a split of this range at key /Table/13 [r5]
I170306 17:26:42.541186 23883 storage/split_queue.go:98  [split,n1,s1,r5/1:/{Table/13-Max},@c421694e00] splitting at key /Table/14/0
I170306 17:26:42.541377 23883 storage/replica_command.go:2456  [split,n1,s1,r5/1:/{Table/13-Max},@c421694e00] initiating a split of this range at key /Table/14 [r6]
I170306 17:26:42.546545 23865 server/server.go:779  [n1] done ensuring all necessary migrations have run
I170306 17:26:42.546646 23865 server/server.go:781  [n1] serving sql connections
W170306 17:26:42.760360 23865 gossip/gossip.go:1178  [n?] no incoming or outgoing connections
W170306 17:26:42.775134 23865 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170306 17:26:42.781524 23865 server/config.go:482  1 storage engine initialized
I170306 17:26:42.783398 23865 server/node.go:443  [n?] store [n0,s0] not bootstrapped
I170306 17:26:42.783493 23865 storage/stores.go:296  [n?] read 0 node addresses from persistent storage
I170306 17:26:42.783613 23865 server/node.go:605  [n?] connecting to gossip network to verify cluster ID...
I170306 17:26:42.857343 24196 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:33637
I170306 17:26:42.858772 24221 gossip/server.go:214  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:39676}
I170306 17:26:42.866494 23865 server/node.go:629  [n?] node connected via gossip and verified as part of cluster "28e50305-28c0-43d3-9e92-a9ea0199a477"
I170306 17:26:42.868031 24190 storage/stores.go:312  [n?] wrote 1 node addresses to persistent storage
I170306 17:26:42.872966 23865 kv/dist_sender.go:368  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170306 17:26:42.881861 23865 server/node.go:334  [n?] new node allocated ID 2
I170306 17:26:42.882186 23865 gossip/gossip.go:293  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:39676" > attrs:<> locality:<> 
I170306 17:26:42.882940 23865 server/node.go:391  [n2] node=2: started with [[]=] engine(s) and attributes []
I170306 17:26:42.883445 23865 sql/executor.go:337  [n2] creating distSQLPlanner with address {tcp 127.0.0.1:39676}
I170306 17:26:42.901636 24223 storage/stores.go:312  [n1] wrote 1 node addresses to persistent storage
I170306 17:26:42.941860 23865 server/server.go:651  [n2] starting https server at 127.0.0.1:60386
I170306 17:26:42.949455 23865 server/server.go:652  [n2] starting grpc/postgres server at 127.0.0.1:39676
I170306 17:26:42.949538 23865 server/server.go:653  [n2] advertising CockroachDB node at 127.0.0.1:39676
I170306 17:26:42.956774 23865 server/server.go:779  [n2] done ensuring all necessary migrations have run
I170306 17:26:42.957031 23865 server/server.go:781  [n2] serving sql connections
I170306 17:26:42.990962 24260 sql/event_log.go:95  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:39676} Attrs: Locality:} ClusterID:28e50305-28c0-43d3-9e92-a9ea0199a477 StartedAt:1488821202882660911 LastUp:1488821202882660911}
W170306 17:26:43.048093 23865 gossip/gossip.go:1178  [n?] no incoming or outgoing connections
W170306 17:26:43.054278 23865 server/status/runtime.go:116  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170306 17:26:43.057039 23865 server/config.go:482  1 storage engine initialized
I170306 17:26:43.080066 23865 server/node.go:443  [n?] store [n0,s0] not bootstrapped
I170306 17:26:43.095888 23865 storage/stores.go:296  [n?] read 0 node addresses from persistent storage
I170306 17:26:43.096344 23865 server/node.go:605  [n?] connecting to gossip network to verify cluster ID...
I170306 17:26:43.121903 24241 server/node.go:586  [n2] bootstrapped store [n2,s2]
I170306 17:26:43.143274 24041 storage/replica_raftstorage.go:414  [replicate,n1,s1,r1/1:/{Min-Table/0},@c420aa7880] generated preemptive snapshot b93b1138 at index 46
I170306 17:26:43.182536 24134 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:33637
I170306 17:26:43.184851 24144 gossip/server.go:214  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:46071}
I170306 17:26:43.189270 23865 server/node.go:629  [n?] node connected via gossip and verified as part of cluster "28e50305-28c0-43d3-9e92-a9ea0199a477"
I170306 17:26:43.190702 24357 storage/stores.go:312  [n?] wrote 1 node addresses to persistent storage
I170306 17:26:43.191442 24357 storage/stores.go:312  [n?] wrote 2 node addresses to persistent storage
I170306 17:26:43.199054 23865 kv/dist_sender.go:368  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170306 17:26:43.207025 23865 server/node.go:334  [n?] new node allocated ID 3
I170306 17:26:43.207466 23865 gossip/gossip.go:293  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:46071" > attrs:<> locality:<> 
I170306 17:26:43.208397 23865 server/node.go:391  [n3] node=3: started with [[]=] engine(s) and attributes []
I170306 17:26:43.208963 23865 sql/executor.go:337  [n3] creating distSQLPlanner with address {tcp 127.0.0.1:46071}
I170306 17:26:43.220472 24387 storage/stores.go:312  [n1] wrote 2 node addresses to persistent storage
I170306 17:26:43.222002 24254 storage/stores.go:312  [n2] wrote 2 node addresses to persistent storage
I170306 17:26:43.255994 23865 server/server.go:651  [n3] starting https server at 127.0.0.1:56040
I170306 17:26:43.256082 23865 server/server.go:652  [n3] starting grpc/postgres server at 127.0.0.1:46071
I170306 17:26:43.256118 23865 server/server.go:653  [n3] advertising CockroachDB node at 127.0.0.1:46071
I170306 17:26:43.272054 24351 sql/event_log.go:95  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:{NetworkField:tcp AddressField:127.0.0.1:46071} Attrs: Locality:} ClusterID:28e50305-28c0-43d3-9e92-a9ea0199a477 StartedAt:1488821203208011943 LastUp:1488821203208011943}
I170306 17:26:43.279009 24348 server/node.go:586  [n3] bootstrapped store [n3,s3]
I170306 17:26:43.294755 23865 server/server.go:779  [n3] done ensuring all necessary migrations have run
I170306 17:26:43.294897 23865 server/server.go:781  [n3] serving sql connections
I170306 17:26:43.329101 24041 storage/store.go:3295  [replicate,n1,s1,r1/1:/{Min-Table/0},@c420aa7880] streamed snapshot: kv pairs: 608, log entries: 36, 63ms
I170306 17:26:43.340531 24466 storage/replica_raftstorage.go:596  [n2,s2,r1/?:{-},@c420734380] applying preemptive snapshot at index 46 (id=b93b1138, encoded size=270218, 1 rocksdb batches, 36 log entries)
I170306 17:26:43.345358 24466 storage/replica_raftstorage.go:604  [n2,s2,r1/?:/{Min-Table/0},@c420734380] applied preemptive snapshot in 5ms [clear=0ms batch=0ms entries=2ms commit=1ms]
I170306 17:26:43.355374 24041 storage/replica_command.go:3312  [replicate,n1,s1,r1/1:/{Min-Table/0},@c420aa7880] change replicas (remove {2 2 2}): read existing descriptor range_id:1 start_key:"" end_key:"\210" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170306 17:26:43.395220 24534 storage/replica.go:2510  [n1,s1,r1/1:/{Min-Table/0},@c420aa7880] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I170306 17:26:43.413839 24505 storage/raft_transport.go:436  [n2] raft transport stream to node 1 established
I170306 17:26:43.415423 24041 storage/replica_raftstorage.go:414  [replicate,n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] generated preemptive snapshot 7a04cc36 at index 20
I170306 17:26:43.549193 24520 storage/replica_raftstorage.go:596  [n3,s3,r4/?:{-},@c423968380] applying preemptive snapshot at index 20 (id=7a04cc36, encoded size=16924, 1 rocksdb batches, 10 log entries)
I170306 17:26:43.550460 24520 storage/replica_raftstorage.go:604  [n3,s3,r4/?:/Table/1{2-3},@c423968380] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170306 17:26:43.568191 24041 storage/store.go:3295  [replicate,n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] streamed snapshot: kv pairs: 30, log entries: 10, 22ms
I170306 17:26:43.570664 24041 storage/replica_command.go:3312  [replicate,n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] change replicas (remove {3 3 2}): read existing descriptor range_id:4 start_key:"\224" end_key:"\225" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170306 17:26:46.250707 24491 storage/replica.go:2510  [n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I170306 17:26:46.268559 24444 storage/raft_transport.go:436  [n3] raft transport stream to node 1 established
I170306 17:26:46.296524 24041 storage/replica_raftstorage.go:414  [replicate,n1,s1,r5/1:/Table/1{3-4},@c421694e00] generated preemptive snapshot 81342359 at index 28
I170306 17:26:46.300235 24041 storage/store.go:3295  [replicate,n1,s1,r5/1:/Table/1{3-4},@c421694e00] streamed snapshot: kv pairs: 52, log entries: 18, 3ms
I170306 17:26:46.301431 24543 storage/replica_raftstorage.go:596  [n3,s3,r5/?:{-},@c422927c00] applying preemptive snapshot at index 28 (id=81342359, encoded size=25380, 1 rocksdb batches, 18 log entries)
I170306 17:26:46.302947 24543 storage/replica_raftstorage.go:604  [n3,s3,r5/?:/Table/1{3-4},@c422927c00] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170306 17:26:46.305851 24041 storage/replica_command.go:3312  [replicate,n1,s1,r5/1:/Table/1{3-4},@c421694e00] change replicas (remove {3 3 2}): read existing descriptor range_id:5 start_key:"\225" end_key:"\226" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170306 17:26:46.334742 24573 storage/replica.go:2510  [n1,s1,r5/1:/Table/1{3-4},@c421694e00] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I170306 17:26:46.339994 24041 storage/replica_raftstorage.go:414  [replicate,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] generated preemptive snapshot 81d83d0a at index 13
I170306 17:26:46.342966 24041 storage/store.go:3295  [replicate,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] streamed snapshot: kv pairs: 10, log entries: 3, 2ms
I170306 17:26:46.344296 24575 storage/replica_raftstorage.go:596  [n2,s2,r6/?:{-},@c423969c00] applying preemptive snapshot at index 13 (id=81d83d0a, encoded size=1720, 1 rocksdb batches, 3 log entries)
I170306 17:26:46.345683 24575 storage/replica_raftstorage.go:604  [n2,s2,r6/?:/{Table/14-Max},@c423969c00] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170306 17:26:46.360191 24041 storage/replica_command.go:3312  [replicate,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] change replicas (remove {2 2 2}): read existing descriptor range_id:6 start_key:"\226" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170306 17:26:46.398692 24607 storage/replica.go:2510  [n1,s1,r6/1:/{Table/14-Max},@c421f7a380] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I170306 17:26:46.405771 24041 storage/replica_raftstorage.go:414  [replicate,n1,s1,r3/1:/Table/1{1-2},@c422a78700] generated preemptive snapshot f8b7d332 at index 18
I170306 17:26:46.428547 24041 storage/store.go:3295  [replicate,n1,s1,r3/1:/Table/1{1-2},@c422a78700] streamed snapshot: kv pairs: 11, log entries: 8, 19ms
I170306 17:26:46.429616 24561 storage/replica_raftstorage.go:596  [n2,s2,r3/?:{-},@c421850700] applying preemptive snapshot at index 18 (id=f8b7d332, encoded size=8523, 1 rocksdb batches, 8 log entries)
I170306 17:26:46.432603 24561 storage/replica_raftstorage.go:604  [n2,s2,r3/?:/Table/1{1-2},@c421850700] applied preemptive snapshot in 3ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170306 17:26:46.440229 24041 storage/replica_command.go:3312  [replicate,n1,s1,r3/1:/Table/1{1-2},@c422a78700] change replicas (remove {2 2 2}): read existing descriptor range_id:3 start_key:"\223" end_key:"\224" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170306 17:26:46.460905 24617 storage/replica.go:2510  [n1,s1,r3/1:/Table/1{1-2},@c422a78700] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2}]
I170306 17:26:46.474717 24041 storage/replica_raftstorage.go:414  [replicate,n1,s1,r2/1:/Table/{0-11},@c422c0ae00] generated preemptive snapshot 8ff23681 at index 28
I170306 17:26:46.483326 24041 storage/store.go:3295  [replicate,n1,s1,r2/1:/Table/{0-11},@c422c0ae00] streamed snapshot: kv pairs: 34, log entries: 18, 8ms
I170306 17:26:46.485102 24631 storage/replica_raftstorage.go:596  [n3,s3,r2/?:{-},@c421839180] applying preemptive snapshot at index 28 (id=8ff23681, encoded size=28906, 1 rocksdb batches, 18 log entries)
I170306 17:26:46.486751 24631 storage/replica_raftstorage.go:604  [n3,s3,r2/?:/Table/{0-11},@c421839180] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170306 17:26:46.495418 24041 storage/replica_command.go:3312  [replicate,n1,s1,r2/1:/Table/{0-11},@c422c0ae00] change replicas (remove {3 3 2}): read existing descriptor range_id:2 start_key:"\210" end_key:"\223" replicas:<node_id:1 store_id:1 replica_id:1 > next_replica_id:2 
I170306 17:26:46.528023 24623 storage/replica.go:2510  [n1,s1,r2/1:/Table/{0-11},@c422c0ae00] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:2}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2}]
I170306 17:26:46.545234 24041 storage/queue.go:693  [n1,replicate] purgatory is now empty
I170306 17:26:46.550084 23884 storage/replica_raftstorage.go:414  [replicate,n1,s1,r3/1:/Table/1{1-2},@c422a78700] generated preemptive snapshot a8e4c29e at index 21
I170306 17:26:46.557882 23884 storage/store.go:3295  [replicate,n1,s1,r3/1:/Table/1{1-2},@c422a78700] streamed snapshot: kv pairs: 12, log entries: 11, 6ms
I170306 17:26:46.563162 24661 storage/replica_raftstorage.go:596  [n3,s3,r3/?:{-},@c4202b8700] applying preemptive snapshot at index 21 (id=a8e4c29e, encoded size=11491, 1 rocksdb batches, 11 log entries)
I170306 17:26:46.565840 24661 storage/replica_raftstorage.go:604  [n3,s3,r3/?:/Table/1{1-2},@c4202b8700] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170306 17:26:46.572277 23884 storage/replica_command.go:3312  [replicate,n1,s1,r3/1:/Table/1{1-2},@c422a78700] change replicas (remove {3 3 3}): read existing descriptor range_id:3 start_key:"\223" end_key:"\224" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I170306 17:26:46.602180 24655 storage/replica.go:2510  [n1,s1,r3/1:/Table/1{1-2},@c422a78700] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I170306 17:26:46.627276 23884 storage/replica_raftstorage.go:414  [replicate,n1,s1,r2/1:/Table/{0-11},@c422c0ae00] generated preemptive snapshot 1fe3a910 at index 31
I170306 17:26:46.655091 23884 storage/store.go:3295  [replicate,n1,s1,r2/1:/Table/{0-11},@c422c0ae00] streamed snapshot: kv pairs: 35, log entries: 21, 20ms
I170306 17:26:46.656761 24449 storage/replica_raftstorage.go:596  [n2,s2,r2/?:{-},@c423bcb180] applying preemptive snapshot at index 31 (id=1fe3a910, encoded size=31874, 1 rocksdb batches, 21 log entries)
I170306 17:26:46.658492 24449 storage/replica_raftstorage.go:604  [n2,s2,r2/?:/Table/{0-11},@c423bcb180] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170306 17:26:46.661647 23884 storage/replica_command.go:3312  [replicate,n1,s1,r2/1:/Table/{0-11},@c422c0ae00] change replicas (remove {2 2 3}): read existing descriptor range_id:2 start_key:"\210" end_key:"\223" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
I170306 17:26:46.700858 24673 storage/replica.go:2510  [n1,s1,r2/1:/Table/{0-11},@c422c0ae00] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
I170306 17:26:46.708003 23884 storage/replica_raftstorage.go:414  [replicate,n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] generated preemptive snapshot d47191f8 at index 25
I170306 17:26:46.721334 23884 storage/store.go:3295  [replicate,n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] streamed snapshot: kv pairs: 32, log entries: 15, 10ms
I170306 17:26:46.728037 24677 storage/replica_raftstorage.go:596  [n2,s2,r4/?:{-},@c423f2a000] applying preemptive snapshot at index 25 (id=d47191f8, encoded size=21100, 1 rocksdb batches, 15 log entries)
I170306 17:26:46.729579 24677 storage/replica_raftstorage.go:604  [n2,s2,r4/?:/Table/1{2-3},@c423f2a000] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170306 17:26:46.733197 23884 storage/replica_command.go:3312  [replicate,n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] change replicas (remove {2 2 3}): read existing descriptor range_id:4 start_key:"\224" end_key:"\225" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
I170306 17:26:46.786630 24524 storage/replica.go:2510  [n1,s1,r4/1:/Table/1{2-3},@c423bd9c00] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
I170306 17:26:46.796004 23884 storage/replica_raftstorage.go:414  [replicate,n1,s1,r1/1:/{Min-Table/0},@c420aa7880] generated preemptive snapshot 40a050a5 at index 74
I170306 17:26:46.847910 23884 storage/store.go:3295  [replicate,n1,s1,r1/1:/{Min-Table/0},@c420aa7880] streamed snapshot: kv pairs: 841, log entries: 64, 51ms
I170306 17:26:46.852600 24527 storage/replica_raftstorage.go:596  [n3,s3,r1/?:{-},@c421ab5c00] applying preemptive snapshot at index 74 (id=40a050a5, encoded size=469844, 1 rocksdb batches, 64 log entries)
I170306 17:26:46.861564 24527 storage/replica_raftstorage.go:604  [n3,s3,r1/?:/{Min-Table/0},@c421ab5c00] applied preemptive snapshot in 8ms [clear=0ms batch=0ms entries=5ms commit=2ms]
I170306 17:26:46.879653 23884 storage/replica_command.go:3312  [replicate,n1,s1,r1/1:/{Min-Table/0},@c420aa7880] change replicas (remove {3 3 3}): read existing descriptor range_id:1 start_key:"" end_key:"\210" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I170306 17:26:46.900122 24724 storage/replica.go:2510  [n1,s1,r1/1:/{Min-Table/0},@c420aa7880] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I170306 17:26:46.914841 23884 storage/replica_raftstorage.go:414  [replicate,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] generated preemptive snapshot eda7f6d3 at index 16
I170306 17:26:46.917195 23884 storage/store.go:3295  [replicate,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] streamed snapshot: kv pairs: 11, log entries: 6, 2ms
I170306 17:26:46.918483 24771 storage/replica_raftstorage.go:596  [n3,s3,r6/?:{-},@c42029aa80] applying preemptive snapshot at index 16 (id=eda7f6d3, encoded size=4695, 1 rocksdb batches, 6 log entries)
I170306 17:26:46.921308 24771 storage/replica_raftstorage.go:604  [n3,s3,r6/?:/{Table/14-Max},@c42029aa80] applied preemptive snapshot in 3ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170306 17:26:46.928876 23884 storage/replica_command.go:3312  [replicate,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] change replicas (remove {3 3 3}): read existing descriptor range_id:6 start_key:"\226" end_key:"\377\377" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:2 store_id:2 replica_id:2 > next_replica_id:3 
I170306 17:26:46.985167 24789 storage/replica.go:2510  [n1,s1,r6/1:/{Table/14-Max},@c421f7a380] proposing ADD_REPLICA {NodeID:3 StoreID:3 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:2 StoreID:2 ReplicaID:2} {NodeID:3 StoreID:3 ReplicaID:3}]
I170306 17:26:46.999586 23884 storage/replica_raftstorage.go:414  [replicate,n1,s1,r5/1:/Table/1{3-4},@c421694e00] generated preemptive snapshot f71c9378 at index 47
I170306 17:26:47.004375 23884 storage/store.go:3295  [replicate,n1,s1,r5/1:/Table/1{3-4},@c421694e00] streamed snapshot: kv pairs: 103, log entries: 37, 4ms
I170306 17:26:47.012637 24819 storage/replica_raftstorage.go:596  [n2,s2,r5/?:{-},@c423923500] applying preemptive snapshot at index 47 (id=f71c9378, encoded size=59834, 1 rocksdb batches, 37 log entries)
I170306 17:26:47.014928 24819 storage/replica_raftstorage.go:604  [n2,s2,r5/?:/Table/1{3-4},@c423923500] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=1ms commit=0ms]
I170306 17:26:47.018092 23884 storage/replica_command.go:3312  [replicate,n1,s1,r5/1:/Table/1{3-4},@c421694e00] change replicas (remove {2 2 3}): read existing descriptor range_id:5 start_key:"\225" end_key:"\226" replicas:<node_id:1 store_id:1 replica_id:1 > replicas:<node_id:3 store_id:3 replica_id:2 > next_replica_id:3 
I170306 17:26:47.069673 24751 storage/replica.go:2510  [n1,s1,r5/1:/Table/1{3-4},@c421694e00] proposing ADD_REPLICA {NodeID:2 StoreID:2 ReplicaID:3}: [{NodeID:1 StoreID:1 ReplicaID:1} {NodeID:3 StoreID:3 ReplicaID:2} {NodeID:2 StoreID:2 ReplicaID:3}]
I170306 17:26:47.382246 24753 sql/event_log.go:95  [client=127.0.0.1:37573,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:bench Statement:CREATE DATABASE bench User:root}
I170306 17:26:47.390722 23883 storage/split_queue.go:98  [split,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] splitting at key /Table/50/0
I170306 17:26:47.393688 23883 storage/replica_command.go:2456  [split,n1,s1,r6/1:/{Table/14-Max},@c421f7a380] initiating a split of this range at key /Table/50 [r7]
I170306 17:26:47.557626 24753 sql/event_log.go:95  [client=127.0.0.1:37573,user=root,n1] Event: "create_table", target: 51, info: {TableName:bench.bank Statement:CREATE TABLE bench.bank (id INT PRIMARY KEY, balance INT, payload STRING, FAMILY (id, balance, payload)) User:root}
I170306 17:26:47.656658 23883 storage/split_queue.go:98  [split,n1,s1,r7/1:/{Table/50-Max},@c422927180] splitting at key /Table/51/0
I170306 17:26:47.656893 23883 storage/replica_command.go:2456  [split,n1,s1,r7/1:/{Table/50-Max},@c422927180] initiating a split of this range at key /Table/51 [r8]
I170306 17:26:47.990370 24753 sql/mon/mem_usage.go:510  [client=127.0.0.1:37573,user=root,n1] txn: memory usage increases to 49 KiB (+49680)
I170306 17:26:47.991008 24753 sql/mon/mem_usage.go:510  [client=127.0.0.1:37573,user=root,n1] txn: memory usage increases to 64 KiB (+140)
I170306 17:26:47.999188 24753 sql/mon/mem_usage.go:510  [client=127.0.0.1:37573,user=root,n1] txn: memory usage increases to 128 KiB (+140)
I170306 17:26:48.294285 24785 storage/replica_command.go:2456  [n1,s1,r8/1:/{Table/51-Max},@c4242d6e00] initiating a split of this range at key /Table/51/1/100 [r9]
I170306 17:26:48.418658 25019 storage/replica_command.go:2456  [n1,s1,r9/1:/{Table/51/1/1���-Max},@c421850380] initiating a split of this range at key /Table/51/1/200 [r10]
I170306 17:26:48.549748 25044 storage/raft_transport.go:436  [n2] raft transport stream to node 3 established
I170306 17:26:48.559213 25032 storage/replica_command.go:2456  [n1,s1,r10/1:/{Table/51/1/2���-Max},@c42099b180] initiating a split of this range at key /Table/51/1/300 [r11]
I170306 17:26:48.560340 25049 storage/raft_transport.go:436  [n3] raft transport stream to node 2 established
I170306 17:26:48.679254 25095 storage/replica_command.go:2456  [n1,s1,r11/1:/{Table/51/1/3���-Max},@c4228b7180] initiating a split of this range at key /Table/51/1/400 [r12]
I170306 17:26:48.802904 25128 storage/replica_command.go:2456  [n1,s1,r12/1:/{Table/51/1/4���-Max},@c4229df180] initiating a split of this range at key /Table/51/1/500 [r13]
I170306 17:26:48.977050 25118 storage/replica_command.go:2456  [n1,s1,r13/1:/{Table/51/1/5���-Max},@c422cc3180] initiating a split of this range at key /Table/51/1/600 [r14]
I170306 17:26:49.185137 25190 storage/replica_command.go:2456  [n1,s1,r14/1:/{Table/51/1/6���-Max},@c422926380] initiating a split of this range at key /Table/51/1/700 [r15]
I170306 17:26:49.317671 25168 storage/replica_command.go:2456  [n1,s1,r15/1:/{Table/51/1/7���-Max},@c4233af880] initiating a split of this range at key /Table/51/1/800 [r16]
I170306 17:26:49.475969 25315 storage/replica_command.go:2456  [n1,s1,r16/1:/{Table/51/1/8���-Max},@c4213cae00] initiating a split of this range at key /Table/51/1/900 [r17]
I170306 17:26:49.949848 25411 storage/replica_command.go:2456  [n1,s1,r17/1:/{Table/51/1/9���-Max},@c4242fc380] initiating a split of this range at key /Table/52/1/500 [r18]
I170306 17:26:50.036491 25475 storage/replica_command.go:2456  [n1,s1,r18/1:/{Table/52/1/5���-Max},@c422307500] initiating a split of this range at key /Table/52/1/800 [r19]
I170306 17:26:50.039212 25361 storage/replica_command.go:2456  [n1,s1,r17/1:/Table/5{1/1/900-2/1/500},@c4242fc380] initiating a split of this range at key /Table/52/1/200 [r20]
I170306 17:26:50.187762 25465 storage/replica_command.go:2456  [n1,s1,r17/1:/Table/5{1/1/900-2/1/200},@c4242fc380] initiating a split of this range at key /Table/52/1/100 [r21]
I170306 17:26:50.197195 25467 storage/replica_command.go:2456  [n1,s1,r20/1:/Table/52/1/{200-500},@c423923180] initiating a split of this range at key /Table/52/1/400 [r22]
I170306 17:26:50.231677 25538 storage/replica_command.go:2456  [n1,s1,r18/1:/Table/52/1/{500-800},@c422307500] initiating a split of this range at key /Table/52/1/700 [r23]
I170306 17:26:50.246556 25539 storage/replica_command.go:2456  [n1,s1,r19/1:/{Table/52/1/8���-Max},@c424018e00] initiating a split of this range at key /Table/52/1/900 [r24]
I170306 17:26:50.395211 25620 storage/replica_command.go:2456  [n1,s1,r20/1:/Table/52/1/{200-400},@c423923180] initiating a split of this range at key /Table/52/1/300 [r25]
I170306 17:26:50.418221 25604 storage/replica_command.go:2456  [n1,s1,r17/1:/Table/5{1/1/900-2/1/100},@c4242fc380] initiating a split of this range at key /Table/52/1 [r26]
I170306 17:26:50.495104 25651 storage/replica_command.go:2456  [n1,s1,r18/1:/Table/52/1/{500-700},@c422307500] initiating a split of this range at key /Table/52/1/600 [r27]
I170306 17:26:50.798126 25764 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader631490072"
I170306 17:26:50.800455 25827 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader118380695"
I170306 17:26:50.805027 25811 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader130643210"
I170306 17:26:50.810061 25826 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader161933537"
I170306 17:26:50.858435 25766 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader171984076"
I170306 17:26:51.002916 25826 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader161933537"
I170306 17:26:51.008339 25827 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader118380695"
I170306 17:26:51.018913 25764 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader631490072"
I170306 17:26:51.020770 25765 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader395862971"
I170306 17:26:51.023915 25784 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader741624798"
I170306 17:26:51.036541 25758 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader120513445"
I170306 17:26:51.052585 25811 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader130643210"
I170306 17:26:51.059598 25657 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader416917696"
I170306 17:26:51.079545 25766 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader171984076"
I170306 17:26:51.090617 25659 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader219587359"
I170306 17:26:51.240182 25765 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader395862971"
I170306 17:26:51.275833 25657 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader416917696"
I170306 17:26:51.326636 25823 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader398854642"
I170306 17:26:51.382804 25777 storage/engine/rocksdb.go:345  opening rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader751199401"
I170306 17:26:51.440321 25784 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader741624798"
I170306 17:26:51.443971 25659 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader219587359"
I170306 17:26:51.459087 25758 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader120513445"
I170306 17:26:51.466896 25973 util/stop/stopper.go:493  quiescing; tasks left:
1      storage/replica_proposal.go:372
1      storage/replica_command.go:1915
1      storage/queue.go:523
I170306 17:26:51.467185 25972 util/stop/stopper.go:493  quiescing; tasks left:
2      storage/replica_proposal.go:372
2      server/node.go:847
I170306 17:26:51.467565 25971 util/stop/stopper.go:493  quiescing; tasks left:
2      storage/replica_command.go:1915
1      storage/queue.go:523
W170306 17:26:51.472681 25049 storage/raft_transport.go:442  [n3] raft transport stream to node 2 failed: rpc error: code = 13 desc = transport is closing
W170306 17:26:51.473025 24500 storage/raft_transport.go:442  [n1] raft transport stream to node 2 failed: rpc error: code = 13 desc = transport is closing
W170306 17:26:51.473558 25044 storage/raft_transport.go:442  [n2] raft transport stream to node 3 failed: EOF
I170306 17:26:51.474158 25809 vendor/google.golang.org/grpc/server.go:752  grpc: Server.processUnaryRPC failed to write status: stream error: code = 1 desc = "context canceled"
I170306 17:26:51.474917 25970 vendor/google.golang.org/grpc/server.go:752  grpc: Server.processUnaryRPC failed to write status: stream error: code = 1 desc = "context canceled"
I170306 17:26:51.475026 25916 vendor/google.golang.org/grpc/server.go:752  grpc: Server.processUnaryRPC failed to write status: connection error: desc = "transport: tls: use of closed connection"
I170306 17:26:51.475274 25972 util/stop/stopper.go:493  quiescing; tasks left:
2      server/node.go:847
1      storage/replica_proposal.go:372
E170306 17:26:51.476353 25807 storage/replica_command.go:1896  [replica consistency checker,n1,s1,r16/1:/Table/51/1/{800-900},@c4213cae00] could not CollectChecksum from replica {2 2 2}: rpc error: code = 13 desc = transport is closing
W170306 17:26:51.477065 24581 storage/raft_transport.go:442  [n1] raft transport stream to node 3 failed: rpc error: code = 13 desc = transport is closing
E170306 17:26:51.477493 25808 storage/replica_command.go:1896  [replica consistency checker,n1,s1,r16/1:/Table/51/1/{800-900},@c4213cae00] could not CollectChecksum from replica {3 3 3}: rpc error: code = 13 desc = transport is closing
W170306 17:26:51.477564 24444 storage/raft_transport.go:442  [n3] raft transport stream to node 1 failed: rpc error: code = 13 desc = transport is closing
W170306 17:26:51.480720 24505 storage/raft_transport.go:442  [n2] raft transport stream to node 1 failed: rpc error: code = 13 desc = transport is closing
E170306 17:26:51.481069 25915 storage/replica_command.go:1896  [replica consistency checker,n3,s3,r12/3:/Table/51/1/{400-500},@c423b0e000] could not CollectChecksum from replica {2 2 2}: rpc error: code = 13 desc = transport is closing
I170306 17:26:51.481951 25973 util/stop/stopper.go:493  quiescing; tasks left:
1      storage/replica_proposal.go:372
1      storage/replica_command.go:1915
I170306 17:26:51.482985 25973 util/stop/stopper.go:493  quiescing; tasks left:
1      storage/replica_command.go:1915
I170306 17:26:51.498095 25972 util/stop/stopper.go:493  quiescing; tasks left:
2      server/node.go:847
E170306 17:26:51.526696 25974 ccl/storageccl/import.go:98  [n2,s2,r26/2:/Table/52/1{-/100},@c423bd8380] writebatch [/Table/52/1/0/0,/Table/52/1/99/0/NULL): context canceled
I170306 17:26:51.526910 25823 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader398854642"
I170306 17:26:51.528233 25972 util/stop/stopper.go:493  quiescing; tasks left:
1      server/node.go:847
E170306 17:26:51.587033 25956 ccl/storageccl/import.go:98  [n2,s2,r27/2:/Table/52/1/{600-700},@c421bb9500] writebatch [/Table/52/1/600/0,/Table/52/1/699/0/NULL): context canceled
I170306 17:26:51.587266 25777 storage/engine/rocksdb.go:410  closing rocksdb instance at "/go/src/github.com/cockroachdb/cockroach/artifacts/RocksDBSstFileReader751199401"
I170306 17:26:51.748616 23839 kv/transport_race.go:71  transport race promotion: ran 48 iterations on up to 832 requests

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/ccl/storageccl/import.go
+++ b/pkg/ccl/storageccl/import.go
@@ -14,7 +14,6 @@ import (
 
 	"golang.org/x/net/context"
 
-	"github.com/cockroachdb/cockroach/pkg/internal/client"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/storage/engine"
@@ -28,28 +27,6 @@ func init() {
 	storage.SetImportCmd(evalImport)
 }
 
-func validateImportRangeEmpty(
-	ctx context.Context, db *client.DB, datarange roachpb.Span, kr KeyRewriter,
-) error {
-	startKey, ok := kr.RewriteKey(append([]byte(nil), datarange.Key...))
-	if !ok {
-		return errors.Errorf("could not rewrite key: %s", startKey)
-	}
-	endKey, ok := kr.RewriteKey(append([]byte(nil), datarange.EndKey...))
-	if !ok {
-		return errors.Errorf("could not rewrite key: %s", endKey)
-	}
-
-	kvs, err := db.Scan(ctx, startKey, endKey, 1)
-	if err != nil {
-		return err
-	}
-	if len(kvs) > 0 {
-		return errors.New("import can only be called on empty ranges")
-	}
-	return nil
-}
-
 // evalImport bulk loads key/value entries.
 func evalImport(ctx context.Context, cArgs storage.CommandArgs) error {
 	args := cArgs.Args.(*roachpb.ImportRequest)
@@ -60,14 +37,6 @@ func evalImport(ctx context.Context, cArgs storage.CommandArgs) error {
 	defer tracing.FinishSpan(span)
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
-
-	// The WriteBatch requests this sends will fail if the ranges they affect
-	// are non-empty, so error early (before the request limiting) if that's the
-	// case. This gets rechecked, so it's okay that this doesn't guarantee that
-	// the range will _stay_ empty.
-	if err := validateImportRangeEmpty(ctx, db, args.DataSpan, kr); err != nil {
-		return err
-	}
 
 	if err := beginLimitedRequest(ctx); err != nil {
 		return err

--- a/pkg/ccl/storageccl/import_test.go
+++ b/pkg/ccl/storageccl/import_test.go
@@ -82,32 +82,28 @@ func TestImport(t *testing.T) {
 	s, _, kvDB := serverutils.StartServer(t, base.TestServerArgs{})
 	defer s.Stopper().Stop()
 
-	req := &roachpb.ImportRequest{
-		Span:        roachpb.Span{Key: reqStartKey},
-		DataSpan:    roachpb.Span{Key: dataStartKey, EndKey: dataEndKey},
-		KeyRewrites: kr,
-		Files: []roachpb.ImportRequest_File{
-			{Dir: storage, Path: sstName},
-		},
-	}
-	b := &client.Batch{}
-	b.AddRawRequest(req)
-	if err := kvDB.Run(ctx, b); err != nil {
-		t.Fatalf("%+v", err)
-	}
-	kvs, err := kvDB.Scan(ctx, reqStartKey, reqEndKey, 0)
-	if err != nil {
-		t.Fatalf("%+v", err)
-	}
-	if len(kvs) != 1 {
-		t.Fatalf("expected 1 kv got %d", len(kvs))
-	}
-
-	// The previous request already inserted data, but the request keyrange is
-	// required to be empty, so running it again should fail.
-	b = &client.Batch{}
-	b.AddRawRequest(req)
-	if err := kvDB.Run(ctx, b); !testutils.IsError(err, "empty ranges") {
-		t.Fatalf("expected 'empty ranges' error got %+v", err)
+	// Import may be retried by DistSender if it takes too long to return, so
+	// make sure it's idempotent.
+	for i := 0; i < 3; i++ {
+		req := &roachpb.ImportRequest{
+			Span:        roachpb.Span{Key: reqStartKey},
+			DataSpan:    roachpb.Span{Key: dataStartKey, EndKey: dataEndKey},
+			KeyRewrites: kr,
+			Files: []roachpb.ImportRequest_File{
+				{Dir: storage, Path: sstName},
+			},
+		}
+		b := &client.Batch{}
+		b.AddRawRequest(req)
+		if err := kvDB.Run(ctx, b); err != nil {
+			t.Fatalf("%+v", err)
+		}
+		kvs, err := kvDB.Scan(ctx, reqStartKey, reqEndKey, 0)
+		if err != nil {
+			t.Fatalf("%+v", err)
+		}
+		if len(kvs) != 1 {
+			t.Fatalf("expected 1 kv got %d", len(kvs))
+		}
 	}
 }

--- a/pkg/ccl/storageccl/writebatch.go
+++ b/pkg/ccl/storageccl/writebatch.go
@@ -28,7 +28,9 @@ func init() {
 	})
 }
 
-// evalWriteBatch applies the operations encoded in a BatchRepr.
+// evalWriteBatch applies the operations encoded in a BatchRepr. Any existing
+// data in the affected keyrange is first cleared (not tombstoned), which makes
+// this command idempotent.
 func evalWriteBatch(
 	ctx context.Context, batch engine.ReadWriter, cArgs storage.CommandArgs, _ roachpb.Response,
 ) (storage.EvalResult, error) {
@@ -64,12 +66,22 @@ func evalWriteBatch(
 	}
 	ms.Add(msBatch)
 
-	// Verify that the key range specified in the request span is empty.
+	// Check if there was data in the affected keyrange. If so, delete it (and
+	// adjust the MVCCStats) before applying the WriteBatch data.
 	iter := batch.NewIterator(false)
 	defer iter.Close()
 	iter.Seek(mvccStartKey)
 	if iter.Valid() && iter.Key().Less(mvccEndKey) {
-		return storage.EvalResult{}, errors.New("WriteBatch can only be called on empty ranges")
+		existingStats, err := iter.ComputeStats(mvccStartKey, mvccEndKey, h.Timestamp.WallTime)
+		if err != nil {
+			return storage.EvalResult{}, err
+		}
+		// TODO(dan): Ideally, this would use `batch.ClearRange` but it doesn't
+		// yet work with read-write batches.
+		if err := batch.ClearIterRange(iter, mvccStartKey, mvccEndKey); err != nil {
+			return storage.EvalResult{}, err
+		}
+		ms.Subtract(existingStats)
 	}
 
 	if err := batch.ApplyBatchRepr(args.Data, false /* !sync */); err != nil {

--- a/pkg/ccl/storageccl/writebatch_test.go
+++ b/pkg/ccl/storageccl/writebatch_test.go
@@ -37,46 +37,67 @@ func TestDBWriteBatch(t *testing.T) {
 	defer s.Stopper().Stop()
 	ctx := context.Background()
 
-	var batch engine.RocksDBBatchBuilder
-	key := engine.MVCCKey{Key: []byte("bb"), Timestamp: hlc.Timestamp{WallTime: 1}}
-	batch.Put(key, roachpb.MakeValueFromString("1").RawBytes)
-	data := batch.Finish()
-
-	if err := db.WriteBatch(ctx, "b", "c", data); err != nil {
-		t.Fatalf("%+v", err)
-	}
-	result, err := db.Get(ctx, "bb")
-	if err != nil {
-		t.Fatalf("%+v", err)
-	}
-	if result := result.ValueBytes(); !bytes.Equal([]byte("1"), result) {
-		t.Errorf("expected \"%s\", got \"%s\"", []byte("1"), result)
+	// Key range in request spans multiple ranges.
+	if err := db.WriteBatch(
+		ctx, keys.LocalMax, keys.MaxKey, nil,
+	); !testutils.IsError(err, "data spans multiple ranges") {
+		t.Fatalf("expected multiple ranges error got: %+v", err)
 	}
 
-	// Key is before the range in the request span.
-	if err := db.WriteBatch(ctx, "d", "e", data); !testutils.IsError(err, "request range") {
-		t.Fatalf("expected request range error got: %+v", err)
-	}
-	// Key is after the range in the request span.
-	if err := db.WriteBatch(ctx, "a", "b", data); !testutils.IsError(err, "request range") {
-		t.Fatalf("expected request range error got: %+v", err)
+	{
+		var batch engine.RocksDBBatchBuilder
+		key := engine.MVCCKey{Key: []byte("bb"), Timestamp: hlc.Timestamp{WallTime: 1}}
+		batch.Put(key, roachpb.MakeValueFromString("1").RawBytes)
+		data := batch.Finish()
+
+		// Key is before the range in the request span.
+		if err := db.WriteBatch(
+			ctx, "d", "e", data,
+		); !testutils.IsError(err, "key not in request range") {
+			t.Fatalf("expected request range error got: %+v", err)
+		}
+		// Key is after the range in the request span.
+		if err := db.WriteBatch(
+			ctx, "a", "b", data,
+		); !testutils.IsError(err, "key not in request range") {
+			t.Fatalf("expected request range error got: %+v", err)
+		}
+
+		if err := db.WriteBatch(ctx, "b", "c", data); err != nil {
+			t.Fatalf("%+v", err)
+		}
+		if result, err := db.Get(ctx, "bb"); err != nil {
+			t.Fatalf("%+v", err)
+		} else if result := result.ValueBytes(); !bytes.Equal([]byte("1"), result) {
+			t.Errorf("expected \"%s\", got \"%s\"", []byte("1"), result)
+		}
 	}
 
 	// Key range in request span is not empty.
-	if err := db.Put(ctx, "cc", 2); err != nil {
-		t.Fatalf("%+v", err)
-	}
-	if err := db.WriteBatch(ctx, "c", "d", nil); !testutils.IsError(err, "empty range") {
-		t.Fatalf("expected empty range error got: %+v", err)
-	}
+	{
+		var batch engine.RocksDBBatchBuilder
+		key := engine.MVCCKey{Key: []byte("bb2"), Timestamp: hlc.Timestamp{WallTime: 1}}
+		batch.Put(key, roachpb.MakeValueFromString("2").RawBytes)
+		data := batch.Finish()
+		if err := db.WriteBatch(ctx, "b", "c", data); err != nil {
+			t.Fatalf("%+v", err)
+		}
 
-	// Key range in request spans multiple ranges.
-	if err := db.WriteBatch(ctx, keys.LocalMax, keys.MaxKey, data); !testutils.IsError(err, "multiple ranges") {
-		t.Fatalf("expected multiple ranges error got: %+v", err)
+		if result, err := db.Get(ctx, "bb2"); err != nil {
+			t.Fatalf("%+v", err)
+		} else if result := result.ValueBytes(); !bytes.Equal([]byte("2"), result) {
+			t.Errorf("expected \"%s\", got \"%s\"", []byte("2"), result)
+		}
+
+		if result, err := db.Get(ctx, "bb"); err != nil {
+			t.Fatalf("%+v", err)
+		} else if result := result.ValueBytes(); result != nil {
+			t.Errorf("expected nil, got \"%s\"", result)
+		}
 	}
 }
 
-func TestWriteBatch(t *testing.T) {
+func TestWriteBatchMVCCStats(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	if !storage.ProposerEvaluatedKVEnabled() {
 		t.Skip("command WriteBatch is not allowed without proposer evaluated KV")
@@ -94,27 +115,54 @@ func TestWriteBatch(t *testing.T) {
 	data := batch.Finish()
 	span := roachpb.Span{Key: []byte("b"), EndKey: []byte("c")}
 
+	// WriteBatch deletes any data that exists in the keyrange before applying
+	// the batch. Put something there to delete. The mvcc stats should be
+	// adjusted accordingly.
+	const numInitialEntries = 100
+	for i := 0; i < numInitialEntries; i++ {
+		if err := e.Put(engine.MVCCKey{Key: append([]byte("b"), byte(i))}, nil); err != nil {
+			t.Fatalf("%+v", err)
+		}
+	}
+
 	cArgs := storage.CommandArgs{
 		Args: &roachpb.WriteBatchRequest{
 			Span:     span,
 			DataSpan: span,
 			Data:     data,
 		},
-		Stats: &enginepb.MVCCStats{},
+		// Start with some stats to represent data throughout the replica's
+		// keyrange.
+		Stats: &enginepb.MVCCStats{
+			LiveBytes: 10000,
+			LiveCount: 10000,
+			KeyBytes:  10000,
+			KeyCount:  10000,
+			ValBytes:  10000,
+			ValCount:  10000,
+		},
 	}
 	if _, err := evalWriteBatch(ctx, e, cArgs, nil); err != nil {
 		t.Fatalf("%+v", err)
 	}
 
-	stats := &enginepb.MVCCStats{
-		LiveBytes: 21,
-		LiveCount: 1,
-		KeyBytes:  15,
-		KeyCount:  1,
-		ValBytes:  6,
-		ValCount:  1,
+	expectedStats := &enginepb.MVCCStats{
+		LiveBytes: 9721,
+		LiveCount: 9901,
+		KeyBytes:  9715,
+		KeyCount:  9901,
+		ValBytes:  10006,
+		ValCount:  10001,
 	}
-	if !reflect.DeepEqual(stats, cArgs.Stats) {
-		t.Errorf("mvcc stats mismatch %+v != %+v", stats, cArgs.Stats)
+	if !reflect.DeepEqual(expectedStats, cArgs.Stats) {
+		t.Errorf("mvcc stats mismatch %+v != %+v", expectedStats, cArgs.Stats)
+	}
+
+	// Run the same WriteBatch command a second time to test the idempotence.
+	if _, err := evalWriteBatch(ctx, e, cArgs, nil); err != nil {
+		t.Fatalf("%+v", err)
+	}
+	if !reflect.DeepEqual(expectedStats, cArgs.Stats) {
+		t.Errorf("mvcc stats mismatch %+v != %+v", expectedStats, cArgs.Stats)
 	}
 }

--- a/pkg/storage/engine/rocksdb.go
+++ b/pkg/storage/engine/rocksdb.go
@@ -913,6 +913,10 @@ func (r *rocksDBBatchIterator) Less(key MVCCKey) bool {
 	return r.iter.Less(key)
 }
 
+func (r *rocksDBBatchIterator) getIter() *C.DBIterator {
+	return r.iter.iter
+}
+
 type rocksDBBatch struct {
 	parent             *RocksDB
 	batch              *C.DBEngine

